### ISSUE TITLE
autocaption_prefix and autocaption_suffix not always added to captions

### ISSUE THREAD

**Issue by erikbrodnick** [replicate/flux-fine-tuner#42]
I observed in the training output logs that the captions did not always include the autocaption_prefix or autocaption_suffix. This is pretty important as it contains the unique trigger token.

After looking at caption.py I noticed this block:

`if autocaption_prefix:
    inp += f"\n\nYou must start the caption with '{autocaption_prefix}'. "

if autocaption_suffix:
    inp += f"\n\nYou must end the caption with '{autocaption_suffix}'."`
    
Instead of relying on the llm to add these which it is clearly failing to do, I suggest manually adding to the resulting output as follows:

`output = self.tokenizer.batch_decode(output_ids, skip_special_tokens=True)[
    0
].strip()

# Manually prepend prefix and append suffix to the generated caption
if autocaption_prefix:
    output = f"{autocaption_prefix} {output}"
if autocaption_suffix:
    output = f"{output} {autocaption_suffix}"

print(f"Caption for {image_path}: {output}")`

**Comment by tomhillable** [comment]
Also have this issue and am seeing only about 50% of the captions having the prefix or suffix enforced.

**Comment by dmund95** [comment]
Yeah I agree. I wanted to make this change and deploy my own model fine-tuning container. I can't find such a guide on replicate. 
Can we not deploy our own fine-tuning logic on replicate ?

**Comment by dmd** [comment]
I am seeing less than 20% of captions having the prefix or suffix.
